agent/proto/resourcesmonitor/resources_monitor.go

```diff
@@ -0,0 +1,130 @@
+package resourcesmonitor
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/agent/proto"
+	"github.com/coder/coder/v2/cli/clistat"
+	"github.com/coder/quartz"
+)
+
+type monitor struct {
+	logger           slog.Logger
+	clock            quartz.Clock
+	config           *proto.GetResourcesMonitoringConfigurationResponse
+	datapointsPusher datapointsPusher
+}
+
+//nolint:revive
+func NewResourcesMonitor(logger slog.Logger, clock quartz.Clock, config *proto.GetResourcesMonitoringConfigurationResponse, datapointsPusher datapointsPusher) *monitor {
+	return &monitor{
+		logger:           logger,
+		clock:            clock,
+		config:           config,
+		datapointsPusher: datapointsPusher,
+	}
+}
+
+type datapointsPusher interface {
+	PushResourcesMonitoringUsage(ctx context.Context, req *proto.PushResourcesMonitoringUsageRequest) (*proto.PushResourcesMonitoringUsageResponse, error)
+}
+
+func (m *monitor) Start(ctx context.Context) error {
+	if !m.config.Enabled {
+		m.logger.Info(ctx, "resources monitoring is disabled - skipping")
+		return nil
+	}
+
+	resourcesFetcher, err := clistat.New()
+	if err != nil {
+		return xerrors.Errorf("failed to create resources fetcher: %w", err)
+	}
+
+	datapointsQueue := NewQueue(int(m.config.Config.NumDatapoints))
+
+	m.clock.TickerFunc(ctx, time.Duration(m.config.Config.CollectionIntervalSeconds*int32(time.Second)), func() error {
```

Comment:
I don't think this duration is doing what we'd expect.

```suggestion
	m.clock.TickerFunc(ctx, time.Duration(m.config.Config.CollectionIntervalSeconds)*time.Second, func() error {
```

I've put together a go playground demonstrating the issue.

https://go.dev/play/p/IDW5tcb5FWS

Reply:
Nice catch here ! Changed 🙈